packages/core/src/loadConfig.ts

```diff
@@ -138,21 +139,23 @@ export async function loadConfig({
 
   let configExport: RsbuildConfigExport;
 
-  if (loader === 'native' || /\.(?:js|mjs|cjs)$/.test(configFilePath)) {
+  // Determine the loading strategy based on the config loader type
+  const useNative = loader === 'native' || loader === 'auto';
```

Comment:
Should we detect that the node version has determined whether the current version supports `native`, when the loader is `auto`?